src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ChannelDistributionUIConfig.java

```diff
@@ -33,40 +33,57 @@
+import com.synopsys.integration.alert.database.api.configuration.model.RegisteredDescriptorModel;
 import com.synopsys.integration.alert.web.model.configuration.FieldModel;
 import com.synopsys.integration.alert.web.model.configuration.FieldValueModel;
 
-@Component
-public class CommonDistributionUIConfig {
+public abstract class ChannelDistributionUIConfig extends UIConfig {
     public static final String KEY_NAME = "channel.common.name";
     public static final String KEY_CHANNEL_NAME = "channel.common.channel.name";
     public static final String KEY_PROVIDER_NAME = "channel.common.provider.name";
     public static final String KEY_FREQUENCY = "channel.common.frequency";
-    private final Logger logger = LoggerFactory.getLogger(CommonDistributionUIConfig.class);
-    private final ConfigurationAccessor configurationAccessor;
+    private final Logger logger = LoggerFactory.getLogger(ChannelDistributionUIConfig.class);
+    private final BaseConfigurationAccessor configurationAccessor;
+    private BaseDescriptorAccessor descriptorAccessor;
 
-    @Autowired
-    public CommonDistributionUIConfig(final ConfigurationAccessor configurationAccessor) {
+    public ChannelDistributionUIConfig(final String label, final String urlName, final String fontAwesomeIcon, final BaseConfigurationAccessor configurationAccessor, final BaseDescriptorAccessor descriptorAccessor) {
+        super(label, urlName, fontAwesomeIcon);
         this.configurationAccessor = configurationAccessor;
+        this.descriptorAccessor = descriptorAccessor;
     }
 
-    public List<ConfigField> createCommonConfigFields(final Set<String> channelDescriptors, final Set<String> providerDescriptors) {
+    @Override
+    public List<ConfigField> createFields() {
         final ConfigField name = TextInputConfigField.createRequired(KEY_NAME, "Name");
         final ConfigField frequency = SelectConfigField.createRequired(KEY_FREQUENCY, "Frequency", Arrays.stream(FrequencyType.values()).map(type -> type.getDisplayName()).collect(Collectors.toList()), this::validateJobName);
-        final ConfigField channelName = SelectConfigField.createRequired(KEY_CHANNEL_NAME, "Channel Type", channelDescriptors);
-        final ConfigField providerName = SelectConfigField.createRequired(KEY_PROVIDER_NAME, "Provider Type", providerDescriptors);
+        final ConfigField channelName = SelectConfigField.createRequired(KEY_CHANNEL_NAME, "Channel Type", getDescriptorNames(DescriptorType.CHANNEL));
+        final ConfigField providerName = SelectConfigField.createRequired(KEY_PROVIDER_NAME, "Provider Type", getDescriptorNames(DescriptorType.PROVIDER));
 
-        return List.of(name, channelName, frequency, providerName);
+        final List<ConfigField> configFields = List.of(name, channelName, frequency, providerName);
+        final List<ConfigField> channelDistributionFields = createChannelDistributionFields();
+        channelDistributionFields.addAll(configFields);
```

Comment:
Same comment here with the addAll as the ProviderUIConfig.  Don't use addAll here since you don't know if createChannelDistributionFields may return an immutable list.